task/cache.py

```diff
@@ -80,7 +86,7 @@ def read(self, resource):
             return None
 
     # Write a resource to the cache in an atomic way
-    def write(self, resource, contents):
+    def write(self, resource: str, contents: Dict[Any, Any]) -> None:
```

Comment:
I've acquired a rather powerful allergy to `Any` during the past year or so... practically the only place I see it as tolerable is on a `**kwargs` used as a passthrough to another function.